src/orion/core/worker/multi_task_algo.py

```diff
@@ -0,0 +1,551 @@
+                self._add_task_id(point, task_id)
+                for point, task_id in zip(new_points, new_task_ids)
+            ]
+            new_trials_with_task_ids = [
+                self._add_task_id(trial, task_id)
+                for trial, task_id in zip(new_trials, new_task_ids)
+            ]
+            # for new_trial, new_trial_with_task_id in zip(new_trials, new_trials_with_task_ids):
+            #     assert False, (new_trial, new_trial_with_task_id)
+
+            log.info(f"About to observe {len(new_points)} new warm-starting points!")
+            # TODO: Not quite sure I understand what the 'results' is supposed to be!
+            # results = [trial.objective for trial in new_trials]
+            results = [
+                {"objective": trial.objective.value, "constraint": [], "gradient": None}
+                for trial in new_trials_with_task_ids
+            ]
+            self.algorithm.observe(new_points_with_task_ids, results)
+
+    def score(self, point: Union[Tuple, np.ndarray]) -> float:
+        """Allow algorithm to evaluate `point` based on a prediction about
+        this parameter set's performance. Return a subjective measure of expected
+        performance.
+
+        By default, return the same score for any parameter (no preference).
+        """
+        assert point in self.space
+        if self._enabled:
+            point = self._add_task_id(point, self.current_task_id)
+        return self.algorithm.score(point)
+
+    def judge(self, point: Point, measurements: Any) -> Optional[Dict]:
+        """Inform an algorithm about online `measurements` of a running trial.
+
+        The algorithm can return a dictionary of data which will be provided
+        as a response to the running environment. Default is None response.
+
+        """
+        assert point in self._space
+        if self._enabled:
+            point = self._add_task_id(point, self.current_task_id)
+        return self.algorithm.judge(
+            self.transformed_space.transform(point), measurements
+        )
+
+    @property
+    def n_suggested(self):
+        """Number of trials suggested by the algorithm **in the target task**"""
+        return len(self.trials_from_target_task)
+
+    @property
+    def n_observed(self):
+        """Number of completed trials observed by the algorithm"""
+        return len(self.trials_from_target_task)
+
+    def set_state(self, state_dict):
+        """Reset the state of the algorithm based on the given state_dict
+
+        :param state_dict: Dictionary representing state of an algorithm
+        """
+        trials_info = state_dict.get("_trials_info")
+        self.unwrapped._trials_info = trials_info
+
+    @property
+    def trials_from_target_task(self) -> Dict[str, Tuple[Tuple[float, ...], Results]]:
+        return {
+            k: (point, results)
+            for k, (point, results) in self.unwrapped._trials_info.items()
+            if self.get_task_id(point) == 0
```

Comment:
Could be a good idea to set a global current_task_id instead of hardcoding it to 0 everywhere.